File: app/src/main/java/com/cyl/musiclake/ui/localmusic/fragment/AlbumFragment.java
package com.cyl.musiclake.ui.localmusic.fragment;

import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.StaggeredGridLayoutManager;
import android.util.Pair;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.cyl.musiclake.R;
import com.cyl.musiclake.data.model.Album;
import com.cyl.musiclake.ui.base.BaseFragment;
import com.cyl.musiclake.ui.common.NavigateUtil;
import com.cyl.musiclake.ui.localmusic.adapter.AlbumAdapter;
import com.cyl.musiclake.ui.localmusic.contract.AlbumsContract;
import com.cyl.musiclake.ui.localmusic.presenter.AlbumPresenter;
import com.cyl.musiclake.ui.common.Constants;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;

/**
 * 功能：本地歌曲列表
 * 作者：yonglong on 2016/8/10 20:49
 * 邮箱：643872807@qq.com
 * 版本：2.5
 */
public class AlbumFragment extends BaseFragment implements AlbumsContract.View {

    @BindView(R.id.recyclerview)
    RecyclerView mRecyclerView;
    @BindView(R.id.tv_empty)
    TextView tv_empty;
    @BindView(R.id.loading)
    LinearLayout loading;
    private AlbumAdapter mAdapter;
    private List<Album> albumList = new ArrayList<>();
    private AlbumPresenter mPresenter;


    public static AlbumFragment newInstance() {

        Bundle args = new Bundle();
        AlbumFragment fragment = new AlbumFragment();
        fragment.setArguments(args);
        return fragment;
    }

    /**
     * 初始化数据
     */
    @Override
    protected void initDatas() {
        mPresenter.loadAlbums("all");
    }


    /**
     * 初始化视图
     *
     * @return
     */
    @Override
    public int getLayoutId() {
        return R.layout.frag_recyclerview;
    }

    /**
     * 初始化控件
     */
    @Override
    public void initViews() {
        mPresenter = new AlbumPresenter(getContext());
        mPresenter.attachView(this);

        mAdapter = new AlbumAdapter(albumList);
        mRecyclerView.setLayoutManager(new StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL));
        mRecyclerView.setAdapter(mAdapter);
        mAdapter.bindToRecyclerView(mRecyclerView);
    }

    @Override
    protected void listener() {
        mAdapter.setOnItemClickListener((adapter, view, position) -> {
            Album album = (Album) adapter.getItem(position);
            NavigateUtil.navigateToAlbum(getActivity(),
                    album.getId(),
                    album.getName(),
                    new Pair<View, String>(view.findViewById(R.id.album), Constants.TRANSTITION_ALBUM));
        });
    }


    @Override
    public void onActivityCreated(final Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        setHasOptionsMenu(true);
    }

    @Override
    public void showLoading() {

    }

    @Override
    public void hideLoading() {
    }

    @Override
    public void showAlbums(List<Album> albumList) {
        mAdapter.setNewData(albumList);
    }

    @Override
    public void showEmptyView() {
        mAdapter.setEmptyView(R.layout.view_song_empty);
    }
}

File: app/src/main/java/com/cyl/musiclake/ui/music/artist/fragment/AlbumFragment.kt
package com.cyl.musiclake.ui.music.artist.fragment

import android.os.Bundle
import androidx.recyclerview.widget.GridLayoutManager
import com.cyl.musiclake.R
import com.cyl.musiclake.bean.Album
import com.cyl.musiclake.bean.Music
import com.cyl.musiclake.common.Extras
import com.cyl.musiclake.ui.base.BaseFragment
import com.cyl.musiclake.ui.music.dialog.BottomDialogFragment
import com.cyl.musiclake.ui.music.local.adapter.AlbumAdapter
import com.cyl.musiclake.ui.music.local.contract.AlbumsContract
import com.cyl.musiclake.ui.music.local.presenter.AlbumPresenter
import kotlinx.android.synthetic.main.fragment_recyclerview_notoolbar.*
import java.util.*

/**
 * 作者：yonglong on 2016/8/15 19:54
 * 邮箱：643872807@qq.com
 * 版本：2.5
 * 专辑
 */
class AlbumFragment : BaseFragment<AlbumPresenter>(), AlbumsContract.View {


    var artistID: Long = 0
    private var transitionName: String? = null
    private var title: String? = null
    private var mAdapter: AlbumAdapter? = null
    private var musicInfos: List<Music> = ArrayList()

    private var bottomDialogFragment: BottomDialogFragment? = null

    override fun loadData() {
        showEmptyState()
        emptyTextView?.text = "功能正在开发中..."
    }

    override fun getLayoutId(): Int {
        return R.layout.fragment_recyclerview_notoolbar
    }

    override fun showAlbums(albumList: MutableList<Album>?) {
        if (albumList?.size != 0) {
            hideLoading()
        }
        albumList?.let {
            mAdapter = AlbumAdapter(it)
            recyclerView.layoutManager = androidx.recyclerview.widget.GridLayoutManager(context, 3, androidx.recyclerview.widget.GridLayoutManager.VERTICAL, false);
            recyclerView.adapter = mAdapter
        }
    }

    public override fun initViews() {
        artistID = arguments?.getLong(Extras.ARTIST_ID) ?: 0L
        transitionName = arguments?.getString(Extras.TRANSITIONNAME)
        title = arguments?.getString(Extras.PLAYLIST_NAME)

    }

    override fun initInjector() {
        mFragmentComponent.inject(this)
    }

    override fun listener() {
        mAdapter?.setOnItemClickListener { _, view, position ->
            if (view.id != R.id.iv_more) {
            }
        }
        mAdapter?.setOnItemChildClickListener { _, _, position ->
        }
    }

    override fun showLoading() {
        super.showLoading()
    }

    override fun hideLoading() {
        super.hideLoading()
    }

    companion object {
        fun newInstance(id: String): AlbumFragment {
            val args = Bundle()
            args.putString(Extras.ARTIST_ID, id)
            val fragment = AlbumFragment()
            fragment.arguments = args
            return fragment
        }
    }
}
